### PR TITLE
Use new base docker images

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

The goal here is to use the new `milmove/milmove-app` docker image for milmove from the transcom/circleci-docker repo. Here are a couple changes:

Go Changes:
- go1.14 now used

Swagger Changes:
- `swagger` is now built into the docker image
- `bin/swagger` is now run directly from the docker image and no longer installed. This is because `go build` doesn't allow you to check `swagger version`. Now we can check `swagger version` which previously would only report `dev` if built locally vs `v0.21.0` if pulled from the docker container.
- Pre-commit hook to validate swagger yaml files is now moved to `scripts/pre-commit-swagger-validate` to ensure it runs both locally and in CircleCI

Go-Bindata (Assets) Changes:
- `go-bindata` is now built into the docker image
- `go-bindata` is now run directly from the docker image and no longer installed due to versioning and release issues with the parent project
- `pkg/assets/assets.go` is removed in circleci and regenerated in the step `check-generated-code` to ensure that code always matches what is checked in.
- `gen-assets` script created to handle asset generation both locally and in CircleCI

AWS Changes:
- `aws` CLI is now upgraded to v2
- `aws-vault` CLI is now upgraded to v5.3

Deployment Changes:
- Repository pushes now use `aws ecr get-login-password` which replaces the old way of logging into docker

Other:
- Updated pre-commit versions

## Setup

Comands to run:

- `make server_test_docker`
- `make e2e_test_docker`
- `make docker_circleci`
- `gen-server` and `make server_generate` or `make pkg/gen/`
- `gen-assets` and `make pkg/assets/assets.go`
- `pre-commit run -a swagger`

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
@reggieriser - I'm attempting to just build the version of go-bindata that we want for the project and then update the code in place. It's not a great solution but we could weather it for a while. You can see some of that in this PR.

File: .circleci/config.yml [hunk 1]
```diff
@@ -344,7 +344,9 @@ jobs:
       - restore_cache:
           keys:
             - go-mod-sources-v2-{{ checksum "go.sum" }}-{{ checksum "scripts/check-go-version" }}
-      - run: echo 'export PATH=${PATH}:~/go/bin:~/transcom/mymove/bin' >> $BASH_ENV
+      - run: echo 'export PATH=~/go/bin:~/transcom/mymove/bin:${PATH}' >> $BASH_ENV
+      - run: make bin/go-bindata
+      - run: rm -rf pkg/assets/assets.go && make pkg/assets/assets.go
```
Comment (chrisgilmerproj) [inline .circleci/config.yml]: @reggieriser - this is the spot. But currently its causing some kind of core-dump that I don't know how to fix just yet. I'll look next week.

**Comment by chrisgilmerproj** [comment]
Use go-bindata 3.18.0 and this should all be resolved now. Requires using the new milmove-app docker image.

**Comment by chrisgilmerproj** [comment]
I'm currently stuck because of failing integration tests on master, otherwise this is ready to go.

This fixes a number of project wide issues for development and deployment. Would like to move forward with this so we can keep iterating on other things.

**Comment by monfresh** [comment]
I tried this as well, and I believe my experience matches Reggie's, except I don't think I saw an error during `make server_test_docker`.

**Comment by chrisgilmerproj** [comment]
>     * `make server_test_docker`: I got a `FAIL` on `github.com/transcom/mymove/pkg/models`.  Looks like it's from: `server_test_1  | === RUN   TestModelSuite/Test_UnpackRateValidation server_test_1  | panic: test timed out after 10m0s`.  I believe everything else passed.

The issue with `make server_test_docker` is somewhat expected. It's resource limited inside docker-compose (which has always been the case) so that we can catch issues that we'd normally only catch on CircleCI.

>     * `make e2e_test_docker`: Everything worked as far as I can tell.

Yes!

>     * `make bin/swagger`, `build-swagger`, `check-swagger-version` -- Are these old?  Commands/scripts not found.

Yep, forgot about these. These are gone.

>     * `make docker_circleci`: Seemed to drop me into the container fine.

Yes!

>     * `gen-server` -- Got `[...]/mymove/scripts/gen-server: line 61: swagger: command not found`, but it eventually finished and I had 17 deleted files under `pkg/gen/primeclient/...`

I forgot to do `swagger_func`, good catch!

>     * `make server_generate`: Didn't fail, but just seemed to check a couple of prereqs.  I then deleted `pkg/gen` and it regenerated and ended up with the 17 deleted files, but also had the same `line 61: swagger: command not found` problem followed by a `make: *** [pkg/gen] Error 127`.

Same issues as above, since `server_generate` calls `gen-server`.

>     * `make pkg/gen/`: said `pkg/gen/` is up to date even after I did a `make clean`

Yes!

>     * `gen-assets`: ran fine, but didn't make any changes (expected).  I deleted `pkg/assets/assets.go` locally and ran again and it regenerated it with no changes (expected).

Yes!

>     * `make pkg/assets/assets.go`: behaved just like `gen-assets`.

Yes!

>     * `pre-commit run -a swagger`: ran fine and passed.

Yes!

> As far as the `assets.go` generation (which I know you wanted me to check out), I believe things worked fine. Just thought I'd check the other things while I was there. (BTW, I did temporarily update my local brew to go 1.14 and aws 2.0.0 for this.)

Looks like you caught my one issue with the client code generation. I'll push a fix soon.

**Comment by chrisgilmerproj** [comment]
> I tried this as well, and I believe my experience matches Reggie's, except I don't think I saw an error during `make server_test_docker`.

This may be because there's either a timing issue or a hardware issue. I find it interesting but otherwise this tool isn't used that often.

**Pull request merged**